### PR TITLE
fix(medusa): Clean response data usage for admin and store fields/expand

### PR DESCRIPTION
**Pull request by adrien2p**
**What**

The clean response data utils was not properly or easy to use in the context of the admin domain. Cleanup and update the req API to include both `storeAllowedProperties` and `adminAllowedProperties` to be used in the appropriate end point handler along side the `cleanResponseData`. As of now, the totals fields of any entities will remain present until the todo gets resolved. Otherwise it means that the user need to pass the total fields manually which in the end will trigger the legacy total computation instead of the performant one

RESOLVES CORE-1162

### CODE REVIEW

File: packages/medusa/src/utils/clean-response-data.ts
Comment (adrien2p) [inline packages/medusa/src/utils/clean-response-data.ts:7]: self-review: did not find a better name for explicitly saying that those props are excluded from being striped out

Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/utils/clean-response-data.ts:7]: typo: `EXCLUDED_FIELDS`

Comment (adrien2p) [reply packages/medusa/src/utils/clean-response-data.ts:7]: ahah thanks @carlos-r-l-rodrigues

File: packages/medusa/src/utils/clean-response-data.ts
Comment (riqwan) [inline packages/medusa/src/utils/clean-response-data.ts:32]: suggestion: Would be better to split this into 2 functions, 1 handling an object and the other handling an array, which reuses the function of handling an object.

This is really confusing to read.

Comment (adrien2p) [reply packages/medusa/src/utils/clean-response-data.ts:32]: Yes good suggestion, ill include that to make it easier to use

File: packages/medusa/src/utils/clean-response-data.ts
Comment (riqwan) [inline packages/medusa/src/utils/clean-response-data.ts:7]: question: is this the right place for this? This should be coming from the place where the route lives.

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (riqwan) [inline packages/medusa/src/api/middlewares/transform-query.ts:116]: thought: I feel the allowed/excluded logic should lie in the routes file and then, as a part of the clean response function, it should accept an excluded or allowed array that will do the exclusion. wdyt?

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (fPolic) [inline packages/medusa/src/api/middlewares/transform-query.ts:48]: question: so we are building `storeAllowedProperties` and  `adminAllowedProperties` in each request

can we use `path` prefix to do: 
`req.allowedProperties = req.path.startsWith("/admin") ? getAdminAllowedProperties() : getStoreAllowedProperties()`

File: packages/medusa/src/utils/clean-response-data.ts
Comment (adrien2p) [reply packages/medusa/src/utils/clean-response-data.ts:7]: It is applicable for everything and is meant to be temporary

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (adrien2p) [reply packages/medusa/src/api/middlewares/transform-query.ts:116]: I just refactored the middleware by creating local utility methods

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (olivermrbl) [reply packages/medusa/src/api/middlewares/transform-query.ts:48]: Alternatively, we could split the middleware into two; one for store and one for admin. Aside from separing the concerns, I don't have an immediate example as to why this would make more sense.
One could imagine, that we add more store/admin specific logic in the future - just as we've done now - and having them separated would them make a lot of sense.

Comment (olivermrbl) [reply packages/medusa/src/api/middlewares/transform-query.ts:48]: ^ this is very non-blocking and just a suggestion

Comment (adrien2p) [reply packages/medusa/src/api/middlewares/transform-query.ts:48]: I completely agree 👌

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (olivermrbl) [inline packages/medusa/src/api/middlewares/transform-query.ts:57]: todo: 
```suggestion
 * Middleware that transform the query input for the store endpoints
```

File: packages/medusa/src/api/middlewares/transform-query.ts
Comment (olivermrbl) [inline packages/medusa/src/api/middlewares/transform-query.ts:12]: praise(this-file): Think the separation of concerns made this super clean - great work!

**Review by olivermrbl — APPROVED** [review]
Great work! Only a minuscule todo :)

**Comment by olivermrbl** [comment]
@adrien2p Don't mind the label here. Added it to later find the PRs that need internal documentation :)

**Comment by olivermrbl** [comment]
@fPolic @riqwan, one of you want to give this another review before we merge?

**Review by fPolic — APPROVED** [review]
LGTM 💪

**Pull request merged**
